agent/engine/docker_task_engine.go

```diff
@@ -302,9 +303,11 @@ func (engine *DockerTaskEngine) monitorExecAgentRunning(ctx context.Context,
 	_, err = engine.execCmdMgr.RestartAgentIfStopped(ctx, engine.client, task, c, dockerID)
 	if err != nil {
 		seelog.Errorf("Task engine [%s]: Failed to restart ExecCommandAgent Process for container [%s]: %v", task.Arn, dockerID, err)
-		// TODO: [ecs-exec] call control plane to report ExecCommandAgent FAIL here
 	}
-	// TODO: [ecs-exec] call control plane to report ExecCommandAgent RESTARTED/UNKNOWN? if RestartIfStopped returns true
+	seelog.Infof("restarted")
+	// whether we restarted or failed to restart, we'll want to emit a state change event
+	// redundant state change events like RUNNING->RUNNING are allowed
+	mTask.emitContainerEvent(mTask.Task, c, "")
```

Comment:
why is this needed in case of a successful restart?

Reply:
we need to emit a container event for either our failure or our success cases here.
If the agent was in STOPPED state, then this would emit RUNNING STSC (which would mean STOPPED -> RUNNING)

Reply:
ah ok, adding a comment will clarify :)